### ISSUE TITLE
Visiting help page loses unpushed changes

### ISSUE THREAD

**Issue by aspiers** [200ok-ch/organice#214]
## Description
Visiting the help page and then returning loses any unpushed changes.

## To reproduce

0. Ensure that the Live sync setting is disabled
1. Go to any file in organice
2. Make a change, so that "Unpushed changes" is visible in the bottom right
3. Click the `?` icon at the top right to visit the help screen
4. Click `Done` at the top right to return from the help screen
5. Observe that unpushed changes have now vanished.

## Expected behavior

Ideally unpushed changes would be preserved after returning from the help.  However if that is not easy to implement, a reasonable alternative short-term fix would be to display "You have unpushed changes - are you sure you want to leave this page?" after clicking the `?` icon.  This is not ideal because it prevents users from visiting the help when they have unpushed changes, but it's a lot better than silently discarding their changes.

## Desktop

 - OS: openSUSE Leap 15.1
 - Browser: Chrome 79

**Comment by munen** [comment]
I confirm. Unfortunately it's not a new bug and the issue is a duplicate (https://github.com/200ok-ch/organice/issues/19).

Because it's a dup, I'm closing this issue.

In any case: Sorry for the inconvenience by this bug!

**Comment by aspiers** [comment]
Actually it's not really a duplicate.  This issue is about visiting the help page, whereas #19 (based on its title, not the description which I see is now a copy'n'paste from here) is about visiting the settings page.  

I did search the issues for `help` before filing this one, and nothing came up.

And in fact the title of #19 seems to be invalid, because if I try the same steps with the settings page, I get the "You have unpushed changes - are you sure you want to leave this page?" dialog.

**Comment by munen** [comment]
> Actually it's not really a duplicate. This issue is about visiting the help page, whereas #19 (based on its title, not the description which I see is now a copy'n'paste from here) is about visiting the settings page.

Well yes, it is. I did actually spend some time looking into this when you posted the issue. It's a nontrivial matter and will require some refactoring. It's a general issue with routing protection.

I added an option on how to proceed in the original issue. Adding more issues for different routes isn't required, because it's a general routing issue.

> And in fact the title of #19 seems to be invalid, because if I try the same steps with the settings page, I get the "You have unpushed changes - are you sure you want to leave this page?" dialog.

That's because it's a nontrivial issue. Sometimes it works, sometimes it doesn't. Same for the help page. Depends on your state.

**Comment by aspiers** [comment]
OK, I think maybe I now understand where we weren't aligned.  You're saying that from an implementation / fix perspective it makes sense to treat them as a single issue, and I can totally appreciate that.  From a user PoV however these appear as separate issues because the behaviour is fundamentally different: in the settings case I'm *always* seeing the "You have unpushed changes - are you sure you want to leave this page?" window, whereas in the help case I'm *always* seeing my unpushed changes silently discarded.

**Comment by munen** [comment]
>  in the settings case I'm always seeing the "You have unpushed changes - are you sure you want to leave this page?" window, whereas in the help case I'm always seeing my unpushed changes silently discarded.

I appreciate that you're saying it's always vs. never. But I read the code and I reproduced both.

It's the same generic problem. That's also why I updated the original issue to state that.

If you really want to prove me wrong on this, make a PR and fix just the one thing. I'll apologize then and give you a well deserved thumbs up. On the other hand, I also won't mind a PR fixing all issues^^  

Until then, I think we've discussed this issue in detail.
